client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -169,6 +169,11 @@ class ProductCreate extends React.Component {
 			// Product type was never switched, so set it before we save.
 			this.props.editProduct( site.ID, product, { type: 'simple' } );
 		}
+
+		if ( ! product.regular_price ) {
+			this.props.editProduct( site.ID, product, { regular_price: '0' } );
```

Comment:
The ActionList part might make this impossible, but do you think it might be better to set some defaults before trying to submit this to the API, rather than firing off another action here in the component? I have some functions in orders that do this kind of thing: [`removeTemporaryIds` & `transformOrderForApi`](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/sites/orders/utils.js#L55). These [run right before the order is handed off](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/sites/orders/actions.js#L100) to the data-layer.